web/src/features/fireWeather/components/graphs/TempRHGraph.tsx

```diff
@@ -5,280 +5,70 @@ import { ObservedValue } from 'api/observationAPI'
-import * as d3Utils from 'utils/d3'
 import { PDT_UTC_OFFSET } from 'utils/constants'
-
-interface WeatherValue {
-  date: Date
-  temp?: number
-  rh?: number
-  modelTemp?: number
-  modelRH?: number
-  forecastTemp?: number
-  forecastRH?: number
-  biasAdjModelTemp?: number
-  biasAdjModelRH?: number
-  hrModelTemp?: number
-  hrModelRH?: number
-  regModelTemp?: number
-  regModelRH?: number
-}
-type ModelSummary = Omit<_ModelSummary, 'datetime'> & { date: Date }
-type ForecastSummary = Omit<_ForecastSummary, 'datetime'> & { date: Date }
-
-interface Props {
+import * as d3Utils from 'utils/d3'
+import * as styles from 'features/fireWeather/components/graphs/TempRHGraph.styles'
+import { ToggleValues } from 'features/fireWeather/components/graphs/useGraphToggles'
+import {
+  getLegendData,
+  useMemoGraphCalculation,
+  WeatherValue
+} from 'features/fireWeather/components/graphs/TempRHGraph.utils'
+
+export interface Props {
+  toggleValues: ToggleValues
   observedValues: ObservedValue[]
-  modelValues: ModelValue[]
-  modelSummaries: _ModelSummary[]
   forecastValues: NoonForecastValue[]
   forecastSummaries: _ForecastSummary[]
-  biasAdjModelValues: ModelValue[]
-  highResModelValues: ModelValue[]
-  highResModelSummaries: _ModelSummary[]
-  regionalModelValues: ModelValue[]
-  regionalModelSummaries: _ModelSummary[]
+  gdpsValues: ModelValue[]
+  gdpsSummaries: _ModelSummary[]
+  biasAdjGdpsValues: ModelValue[]
+  hrdpsValues: ModelValue[]
+  hrdpsSummaries: _ModelSummary[]
+  rdpsValues: ModelValue[]
+  rdpsSummaries: _ModelSummary[]
 }
 
-const TempRHGraph: React.FunctionComponent<Props> = ({
-  observedValues: _observedValues,
-  modelValues: _modelValues,
-  modelSummaries: _modelSummaries,
-  forecastValues: _forecastValues,
-  forecastSummaries: _forecastSummaries,
-  biasAdjModelValues: _biasAdjModelValues,
-  highResModelValues: _highResModelValues,
-  highResModelSummaries: _highResModelSummaries,
-  regionalModelValues: _regionalModelValues,
-  regionalModelSummaries: _regionalModelSummaries
-}: Props) => {
+/* Table layout constants */
+const margin = { top: 10, right: 40, bottom: 20, left: 40 }
+const graphWidth = 600
+const graphHeight = 300
+const legendHeight = 15
+const chartWidth = graphWidth - margin.left - margin.right
+const chartHeight = graphHeight - margin.top - margin.bottom - 50
+
+const TempRHGraph: React.FunctionComponent<Props> = (props: Props) => {
+  const { toggleValues } = props
   const classes = styles.useStyles()
-  const svgRef = useRef(null)
+  const utcOffset = PDT_UTC_OFFSET
 
-  // Keep track of the sidebar position & size so that the sidebar can stay
-  // exactly as it is when toggling weather sources
-  // Use mutable ref object that doesn't cause re-render
-  const brushSelection = useRef<[number, number] | null>(null)
+  const graphRef = useRef<SVGSVGElement>(null)
+  const legendRef = useRef<SVGSVGElement>(null)
+  const gc = useMemoGraphCalculation(props, chartWidth)
```

Comment:
I'd prefer to see `gc` renamed to `graphCalc` or something similar to make it more obvious what it is.